sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/util/RegexUtil.java

```diff
@@ -23,8 +26,16 @@ private RegexUtil() {}
    * @return true if string matches the pattern, otherwise false
    */
   public static boolean compare(String expected, String actual) {
-    String expectedPattern = reformatEscapeCharacter(expected);
-    return Pattern.compile(expectedPattern).matcher(Objects.requireNonNull(actual)).find();
+    boolean result;
+    if (areBothEmpty(actual, expected)) {
+      result = true;
+    } else if (areBothDefined(actual, expected)) {
+      String expectedPattern = reformatEscapeCharacter(expected);
+      result = Pattern.compile(expectedPattern).matcher(Objects.requireNonNull(actual)).find();
+    } else {
+      result = false;
+    }
+    return result;
```

Comment:
Can you remove the _result_ variable?
Doing something like:
```
if (bothEmpty) return true
if (bothDefined) return Pattern.compile
return false
```
seems more clear to me than looking at the result variable and than wating for return at the end,